tests/ulm-with-contract/types.str-raw.run

```diff
@@ -0,0 +1,21 @@
+push "strEndpoint"; 
+hold_string_from_test_stack; 
+hold_list_values_from_test_stack; 
+hold_list_values_from_test_stack; 
+encode_call_data; 
+
+return_value; 
+mock CallData; 
+
+call_contract 12345;
+return_value;
+check_eq ();
+
+push_status; 
+check_eq 2; 
+
+check_raw_output
+    b"\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x20" +
+    b"\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x05" +
+    b"Hello\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00"
```

Comment:
Based on my understanding of how encoded values work, this seems like it has an extra value in it?

I expect that the endpoint, when  a str is encoded as:

```
| 32: str.len() | 32x: str payload |
```

where:

1.  the notation `| n: payload |` describes a byte field of length `n` with the given payload
2.  `x` is chosen to be the smallest value of `x` such that `32x > str.len()`.

I don't understand where the initial value x20 is coming from.

Reply:
See the answer here: https://github.com/Pi-Squared-Inc/rust-demo-semantics/pull/185#discussion_r1849089863 , but a single string is encoded like this (when part of a request, response, log entry, and probably other stuff):
```
| 32: prefix-length | 32: str.len() | 32x: str payload |
|-------prefix------|-------------suffix---------------|
```